src/rvoice/fluid_rvoice.c

```diff
@@ -319,11 +319,32 @@ fluid_rvoice_write (fluid_rvoice_t* voice, fluid_real_t *dsp_buf)
    * buffer. It is the ratio between the frequencies of original
    * waveform and output waveform.*/
   voice->dsp.phase_incr = fluid_ct2hz_real(voice->dsp.pitch + 
+	  voice->dsp.pitchoffset +
      fluid_lfo_get_val(&voice->envlfo.modlfo) * voice->envlfo.modlfo_to_pitch
      + fluid_lfo_get_val(&voice->envlfo.viblfo) * voice->envlfo.viblfo_to_pitch
      + fluid_adsr_env_get_val(&voice->envlfo.modenv) * voice->envlfo.modenv_to_pitch) 
      / voice->dsp.root_pitch_hz;
 
+  /******************* portamento ****************/
+  /* pitchoffset is updated if enabled.
+     Pitchoffset will be added to dsp pitch at
+	 phase calculation time */
+  if (voice->dsp.pitchinc > 0.0f)
+  {	/* portamento is enabled, so update pitchoffset */
+	voice->dsp.pitchoffset += voice->dsp.pitchinc;
+	/* when pitchoffset reaches 0.0f, portamento is disabled */
+	if (voice->dsp.pitchoffset > 0.0f) 
+		voice->dsp.pitchoffset = voice->dsp.pitchinc = 0.0f;
+  }
+  else if (voice->dsp.pitchinc < 0.0f)
+  {	/* portamento is enabled, so update pitchoffset */
+	voice->dsp.pitchoffset += voice->dsp.pitchinc;
+	/* when pitchoffset reaches 0.0f, portamento is disabled */
+	if (voice->dsp.pitchoffset < 0.0f) 
+		voice->dsp.pitchoffset = voice->dsp.pitchinc = 0.0f;
+  }
```

Comment:
Couldn't this whole portamento pitch calc. be simplified to:

```
voice->dsp.pitchoffset += voice->dsp.pitchinc;

if ((voice->dsp.pitchinc > 0.0f && voice->dsp.pitchoffset > 0.0f) ||
    (voice->dsp.pitchinc < 0.0f && voice->dsp.pitchoffset < 0.0f))
  voice->dsp.pitchoffset = voice->dsp.pitchinc = 0.0f;
```

Reply:
yes it could, it will be done.